### ISSUE TITLE
Number getter use same api as Text

### ISSUE THREAD

**Issue by neithanmo** [Zondax/icp-client-cpp#23]
The getNumber getter for IdlValue uses the same functions as getText.
but Number is a different variant in the original IDLValue definition.
bellow current implementation;
```
std::string IdlValue::getNumber() {
  if (ptr == nullptr) {
    std::cerr << "IdlValue instance uninitialized" << std::endl;
    return "";
  }
  CText *ctext = text_from_idl_value(ptr);
  const char *str = ctext_str(ctext);
  uintptr_t len = ctext_len(ctext);

  std::string text(str, len);

  ctext_destroy(ctext);

  return text;
}
```
and the **text_from_idl_value** implementations is:
```
#[no_mangle]
pub extern "C" fn text_from_idl_value(ptr: &IDLValue) -> Option<Box<CText>> {
    let s = match ptr {
        IDLValue::Text(v) => v.to_owned(),
        _ => return None,
    };

    let c_string = CString::new(s).ok()?;
    let data = c_string.into_bytes();

    let c_text = Box::new(CText { data });
    Some(c_text)
}
```
so this means, it is not correct to try to retrieve a Number variant from the IDLValue enum using the CText api, as it will always fail.

:link: [zboto Link](https://app.clickup.com/t/861n00hk9)

**Comment by chcmedeiros** [comment]
Yes that is not correct, although Number Type is retrieving a number in string, we should be using the number rust interface. This was a copy and paste error, I think.

**Comment by becominginsane** [comment]
This is internally represented as a "BigNum" https://docs.rs/candid/latest/candid/types/number/index.html and from the docs it also seems to be using LEB128 encoding

**Comment by neithanmo** [comment]
fixed via #24
